packages/@aws-cdk/aws-ec2/lib/vpn.ts

```diff
@@ -292,7 +292,6 @@ export class VpnConnection extends VpnConnectionBase {
     if (!props.vpc.vpnGatewayId) {
       props.vpc.enableVpnGateway({
         type: 'ipsec.1',
-        amazonSideAsn: props.asn,
```

Comment:
This is definitely a breaking change. Please add a feature flag for this and add a new optional property, `amazonSideAsn`, to the `VpnConnectionProps`. Take a look at #20765 for an example of how to add feature flags. 

Then this becomes 

```
     props.vpc.enableVpnGateway({
        type: 'ipsec.1',
        amazonSideAsn,
```